File: mobile/src/types/travelExpenseReport.ts
export type TravelExpensePersonType = "therapist" | "doctor";

export type TravelExpenseReportFormat = "pdf" | "xlsx" | "csv";

export interface TravelExpenseReportFilters {
  personType?: TravelExpensePersonType;
  personId?: number | "all";
  month?: string;
  startDate?: string;
  endDate?: string;
}

export interface TravelExpenseReportRow {
  date: string;
  patient_name: string;
  from_address: string;
  to_address: string;
  km: number;
  fare: number;
  daily_allowance: number;
  others: number;
  total: number;
}

export interface TravelExpenseReportGroup {
  person_id: number;
  person_name: string;
  rows: TravelExpenseReportRow[];
  total_km: number;
  total_fare: number;
  total_daily_allowance: number;
  total_others: number;
  grand_total: number;
}

export interface TravelExpenseReportResponse {
  heading: string;
  person_type: TravelExpensePersonType;
  scope: "individual" | "all";
  person_id: number | null;
  person_name: string | null;
  period_label: string;
  start_date: string;
  end_date: string;
  groups: TravelExpenseReportGroup[];
  total_km: number;
  total_fare: number;
  total_daily_allowance: number;
  total_others: number;
  grand_total: number;
  row_count: number;
  generated_at: string;
  warnings: string[];
}

File: backend/app/schemas/travel_expense_report.py
from datetime import date, datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


PersonType = Literal["therapist", "doctor"]


class TravelExpenseReportRow(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    date: date
    patient_name: str
    from_address: str
    to_address: str
    km: float
    fare: float
    daily_allowance: float
    others: float
    total: float


class TravelExpensePersonGroup(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    person_id: int
    person_name: str
    rows: list[TravelExpenseReportRow]
    total_km: float
    total_fare: float
    total_daily_allowance: float
    total_others: float
    grand_total: float


class TravelExpenseReportResponse(BaseModel):
    heading: str
    person_type: PersonType
    scope: Literal["individual", "all"]
    person_id: int | None = None
    person_name: str | None = None
    period_label: str
    start_date: date
    end_date: date
    groups: list[TravelExpensePersonGroup]
    total_km: float
    total_fare: float
    total_daily_allowance: float
    total_others: float
    grand_total: float
    row_count: int
    generated_at: datetime
    warnings: list[str] = Field(default_factory=list)
